docs: mention the `inject` method in the DI guide (#50942)
angular/angular: aio/content/guide/dependency-injection.md

}
</code-example>

> Another way to inject a dependency without defining the constructor is to use the [inject](api/core/inject) method.

Reviewer: ```suggestion
Another option is to use the [inject](api/core/inject) method:
```

Author: Is it fine if I just apply your suggestion (we end up with two commits unsquashed)? I don’t have a laptop until next week to squash the commits but I don’t want to halt the progress on the PR.

Reviewer: @ToneyPK yes, you can try doing this and use a commit message like this: `docs: simplify a note in DI guide` (so the `lint` CI job passes). Please let me know if that worked. Thank you.

Author: @AndrewKushnir I applied the commit with your message and the lint job has passed successfully.

<code-example language="typescript">
@Component({ … })
class HeroListComponent {

>   private service: inject(HeroService);

Reviewer 2: ```
private service = inject(HeroService);
```

Author: good catch, thanks.
corrected

}

> </code-example>

Reviewer: @ToneyPK thanks for improving Angular docs!

I think this change would also benefit from a link to a guide that explains what an injection context is (i.e. where the `inject` function can be used). The mentioned guide is in a different PR, see #49782. Marking this PR as blocked for now (blocked by PR #49782).

Author: @AndrewKushnir My pleasure.

Yes, even a better idea!

Reviewer: The mentioned PR is now merged and it also added a reference to the injection context into the `inject` docs, so we can omit it here. I've just left a minor comment on how we can simplify the wording a bit.
